### PR TITLE
Add EtherInc (ETI) Support

### PR DESCRIPTION
**Pull request by suryanshkumr**
# Description
EtherInc is a fork of Ethereum to power the future of blockchain based organizations.

* Homepage: https://einc.io/
* Node: https://api.einc.io/jsonrpc/mainnet
* Explorer: https://explorer.einc.io/
* BIP44 / Chainid 101 ( satoshilabs/slips#438 )

# Changes
This PR adds EtherInc node support

### CODE REVIEW

File: app/scripts/controllers/decryptWalletCtrl.js [hunk 1]
```diff
@@ -40,7 +40,8 @@ var decryptWalletCtrl = function($scope, $sce, walletService) {
     hwAtheiosPath: "m/44'/1620'/0'/0", // first address: m/44'/1620'/0'/0/0
     hwIolitePath: "m/44'/1171337'/0'/0", // first address: m/44'/1171337'/0'/0/0
     tomoPath: "m/44'/889'/0'/0", // first address: m/44'/889'/0'/0/0
-    hwMixPath: "m/44'/76'/0'/0" // first address: m/44'/76'/0'/0/0
+    hwMixPath: "m/44'/76'/0'/0", // first address: m/44'/76'/0'/0/0
+    etiPath: "m/44'/60'/0'/0", // first address: m/44'/76'/0'/0/0
```
Comment (hackmod) [inline app/scripts/controllers/decryptWalletCtrl.js]: wrong HD path.

File: app/scripts/directives/walletDecryptDrtv.html [hunk 2]
```diff
@@ -1178,6 +1178,20 @@ <h4 id="modalTitle" class="modal-title" translate="ADD_Radio_5_Path" style="marg
...
                </label>
             </div>
+            
+            <div class="col-sm-4">
+              <label class="radio small">
+                <input aria-describedby="Path: TREZOR (ETI) {{HDWallet.hwIolitePath}}"
+                        ng-change="onHDDPathChange()"
+                        ng-model="HDWallet.dPath"
+                        type="radio"
+                       value="{{HDWallet.hwIolitePath}}"/>
```
Comment (hackmod) [inline app/scripts/directives/walletDecryptDrtv.html]: wrong path. it's not Iolite

File: app/styles/etherwallet-custom.less [hunk 3]
```diff
@@ -692,6 +693,9 @@ label small {
   li:nth-child(32) {
     border-left: 2px solid @brand-ath;
   }
+  li:nth-child(33) {
```
Comment (hackmod) [inline app/styles/etherwallet-custom.less]: wrong numbering

**Comment by hackmod** [comment]
- chainId `101` - https://github.com/etherinc/go-etherinc/blob/master/params/config.go#L34
- bip44 `464` - https://github.com/satoshilabs/slips/blame/master/slip-0044.md#L495

**Comment by mkrufky** [comment]
blocked by https://github.com/kvhnuke/etherwallet/issues/2145

File: app/scripts/controllers/decryptWalletCtrl.js [hunk 4]
```diff
@@ -107,6 +108,9 @@ var decryptWalletCtrl = function($scope, $sce, walletService) {
         case nodes.nodeTypes.MIX:
           $scope.HDWallet.dPath = $scope.HDWallet.hwMixPath;
           break;
+        case nodes.nodeTypes.ETI:
+          $scope.HDWallet.dPath = $scope.HDWallet.etiPath;
+          break;
         default:
```
Comment (mkrufky) [inline app/scripts/controllers/decryptWalletCtrl.js:114]: Please link to your Ledger PR, or remove the Ledger support if it isn't actually supported.

File: app/scripts/controllers/decryptWalletCtrl.js [hunk 5]
```diff
@@ -175,6 +179,9 @@ var decryptWalletCtrl = function($scope, $sce, walletService) {
         case nodes.nodeTypes.TOMO:
           $scope.HDWallet.dPath = $scope.HDWallet.tomoPath;
           break;
+        case nodes.nodeTypes.ETI:
+          $scope.HDWallet.dPath = $scope.HDWallet.etiPath;
+          break;
         default:
```
Comment (mkrufky) [inline app/scripts/controllers/decryptWalletCtrl.js:185]: Please link to your Trezor PR, or remove the Trezor support if it isn't actually supported.

File: app/scripts/directives/walletDecryptDrtv.html [hunk 6]
```diff
@@ -47,7 +47,7 @@ <h4 translate="decrypt_Access">
     <!-- Ledger -->
     <label aria-flowto="aria3"
            class="radio"
-           ng-show="ajaxReq.type=='ETH'||ajaxReq.type=='ETC'||ajaxReq.type=='ROPSTEN ETH'||ajaxReq.type=='RINKEBY ETH'||ajaxReq.type=='KOVAN ETH'||ajaxReq.type=='EXP'||ajaxReq.type=='UBQ'||ajaxReq.type=='POA'||ajaxReq.type=='TOMO'||ajaxReq.type=='ESN'||ajaxReq.type=='AKA'||ajaxReq.type=='PIRL'||ajaxReq.type=='ETHO'||ajaxReq.type=='EGEM'||ajaxReq.type=='CLO'||ajaxReq.type=='ATH'||ajaxReq.type=='MUSIC'||ajaxReq.type=='EOSC'||ajaxReq.type=='GO'||ajaxReq.type=='MIX'||ajaxReq.type=='ILT'">
+           ng-show="ajaxReq.type=='ETH'||ajaxReq.type=='ETC'||ajaxReq.type=='ROPSTEN ETH'||ajaxReq.type=='RINKEBY ETH'||ajaxReq.type=='KOVAN ETH'||ajaxReq.type=='EXP'||ajaxReq.type=='UBQ'||ajaxReq.type=='POA'||ajaxReq.type=='TOMO'||ajaxReq.type=='ESN'||ajaxReq.type=='AKA'||ajaxReq.type=='PIRL'||ajaxReq.type=='ETHO'||ajaxReq.type=='EGEM'||ajaxReq.type=='CLO'||ajaxReq.type=='ATH'||ajaxReq.type=='MUSIC'||ajaxReq.type=='EOSC'||ajaxReq.type=='GO'||ajaxReq.type=='MIX'||ajaxReq.type=='ILT'||ajaxReq.type=='ETI'">
```
Comment (mkrufky) [inline app/scripts/directives/walletDecryptDrtv.html:50]: Please link to your Ledger PR, or remove the Ledger support if it isn't actually supported.

File: app/scripts/directives/walletDecryptDrtv.html [hunk 7]
```diff
@@ -59,7 +59,7 @@ <h4 translate="decrypt_Access">
     <!-- TREZOR -->
     <label class="radio"
            aria-flowto="aria4"
-           ng-show="ajaxReq.type=='ETH'||ajaxReq.type=='ETC'||ajaxReq.type=='ROPSTEN ETH'||ajaxReq.type=='RINKEBY ETH'||ajaxReq.type=='KOVAN ETH'||ajaxReq.type=='EXP'||ajaxReq.type=='UBQ'||ajaxReq.type=='POA'||ajaxReq.type=='TOMO'||ajaxReq.type=='ELLA'||ajaxReq.type=='EGEM'||ajaxReq.type=='CLO'||ajaxReq.type=='ETSC'||ajaxReq.type=='MUSIC'||ajaxReq.type=='GO'||ajaxReq.type=='EOSC'||ajaxReq.type=='AKA'||ajaxReq.type=='ESN'||ajaxReq.type=='PIRL'||ajaxReq.type=='ETHO'||ajaxReq.type=='ATH'||ajaxReq.type=='ILT'">
+           ng-show="ajaxReq.type=='ETH'||ajaxReq.type=='ETC'||ajaxReq.type=='ROPSTEN ETH'||ajaxReq.type=='RINKEBY ETH'||ajaxReq.type=='KOVAN ETH'||ajaxReq.type=='EXP'||ajaxReq.type=='UBQ'||ajaxReq.type=='POA'||ajaxReq.type=='TOMO'||ajaxReq.type=='ELLA'||ajaxReq.type=='EGEM'||ajaxReq.type=='CLO'||ajaxReq.type=='ETSC'||ajaxReq.type=='MUSIC'||ajaxReq.type=='GO'||ajaxReq.type=='EOSC'||ajaxReq.type=='AKA'||ajaxReq.type=='ESN'||ajaxReq.type=='PIRL'||ajaxReq.type=='ETHO'||ajaxReq.type=='ATH'||ajaxReq.type=='ILT'||ajaxReq.type=='ETI'">
```
Comment (mkrufky) [inline app/scripts/directives/walletDecryptDrtv.html:62]: Please link to your Trezor PR, or remove the Trezor support if it isn't actually supported.

**Review by mkrufky — CHANGES_REQUESTED** [review]
As described in https://github.com/kvhnuke/etherwallet/issues/2145 , both EtherInc and WebChain Networks are using Chain ID `101` , which is very problematic for user security. Given the circumstances, it is unsafe to use either of these chains without a unique chain ID for proper replay attack protection.

**Comment by hackmod** [comment]
and do not insert your chain in the middle of the list. just append your network at the end.

**Comment by gamalielhere** [comment]
@suryanshkumr see @mkrufky's comments. Will wait till those are resolved.

**Comment by tayvano** [comment]
We appreciate you creating this pull request but upon further investigation it looks like you share a chainid with another chain (see: https://github.com/MyCryptoHQ/MyCrypto/pull/2217)

In case you don't know, if two chains have the same id, replay attacks are possible which means a user could inadvertently have their funds spent on both chains. This occurred during the ETH/ETC split and was unfortunate for all users innocently trying to move funds. 

MyCrypto prioritizes user experience and user safety and we cannot include EtherInc (or WebChain) on our platform until you have differing chainids.

We don't have the desire nor willingness to moderate a discussion between the two of you or be in the middle so I strongly recommend getting together with the WebChain team independently and figuring out a resolution. In the long run, working together to resolve this will make both of your products stronger, more secure, and show good faith towards your individual communities and the larger blockchain community. If you do not, integrations with products and exchanges will continue to be problematic.

Again, we appreciate the PR and look forward to merging you when the chainid issue is resolved. Thanks for your contributions to this space & I hope the bearish market doesn't have you or your team too down. Much love. 💖

**Comment by tayvano** [comment]
Hey MEW team...just a heads up but it looks like you may have merged WebChain (which shares the same chainid as EtherInc).

We're taking the following stance on the matter for both WebChain and EtherInc and merging neither until it's resolved. Adding one and not the other is only slightly less dangerous than adding both until their chainids are different.

> We appreciate you creating this pull request but upon further investigation it looks like you share a chainid with another chain (see: https://github.com/MyCryptoHQ/MyCrypto/pull/2217)

> In case you don't know, if two chains have the same id, replay attacks are possible which means a user could inadvertently have their funds spent on both chains. This occurred during the ETH/ETC split and was unfortunate for all users innocently trying to move funds. 

> MyCrypto prioritizes user experience and user safety and we cannot include WebChain (or EtherInc) on our platform until you have differing chainids.

> We don't have the desire nor willingness to moderate a discussion between the two of you or be in the middle so I strongly recommend getting together with the EtherInc team independently and figuring out a resolution. In the long run, working together to resolve this will make both of your products stronger, more secure, and show good faith towards your individual communities and the larger blockchain community. If you do not, integrations with products and exchanges will continue to be problematic.

> Again, we appreciate the PR and look forward to merging you when the chainid issue is resolved. Thanks for your contributions to this space & I hope the bearish market doesn't have you or your team too down. Much love. 💖

**Comment by kvhnuke** [comment]
Out of all possibilities, I can't believe chains are still arguing about chainIDs. We will look into this as well, thanks for notifying

**Pull request closed without merging**